src/Microsoft.ML.Data/DataLoadSave/CompositeDataLoader.cs

```diff
@@ -110,5 +113,25 @@ private static VersionInfo GetVersionInfo()
                 loaderSignature: LoaderSignature,
                 loaderAssemblyName: typeof(CompositeDataLoader<,>).Assembly.FullName);
         }
+
+        #region IDisposable Support
+        private bool _disposed;
+
+        private void Dispose(bool disposing)
+        {
+            if (_disposed)
+                return;
+
+            if (disposing)
+                Transformer.Dispose();
+
+            _disposed = true;
+        }
+
+        void IDisposable.Dispose()
```

Comment:
Why are you explicitly implementing `IDisposable`? Instead, you can just have a `public void Dispose()` method here.

The reason to not explicitly implement an interface is so users of the class can just call `Dispose()` as necessary. They don't need to cast to `IDisposable` to call it.